[exporterhelper] Allow to specify max queue size in bytes

aboguszewski-sumo opened the pull request.

aboguszewski-sumo committed:
[exporterhelper] Allow to specify max queue size in bytes

swiatekm commented on exporter/exporterhelper/queued_retry_experimental.go:
```diff
@@ -219,11 +237,19 @@ func (qrs *queuedRetrySender) start(ctx context.Context, host component.Host) er
+			return fmt.Errorf("failed to create retry queue capacity in batches metric: %w", err)
+		}
+
 		err = globalInstruments.queueCapacity.UpsertEntry(func() int64 {
```
Shouldn't this be a separate metric?

aboguszewski-sumo replied:
You're right, this is what copypasting leads to.

swiatekm commented on exporter/exporterhelper/internal/persistent_storage_batch.go:
```diff
@@ -59,15 +59,16 @@ func (bof *batchStruct) execute(ctx context.Context) (*batchStruct, error) {
 }
 
 // set adds a Set operation to the batch
-func (bof *batchStruct) set(key string, value interface{}, marshal func(interface{}) ([]byte, error)) *batchStruct {
```
Instead of making all these functions explicitly return the size, can't we just add a method to `batchStruct` that would calculate the total data size? Data in a batchstruct is already marshalled, so it'd be just iterating over operations.

swiatekm reviewed (commented):
What about the in-memory queue? Do you want to only make the `queue_size` => `queue_size_batches` change for the persistent queue? That would be really confusing,

aboguszewski-sumo committed:
[exporterhelper] Deprecate queue_size configuration option
The capacity of the queue can now be expressed in bytes. Because of that, parameter queue_size has been renamed to queue_size_batches.

aboguszewski-sumo commented:
I just forgot about it. I changed it for now, but it will still need some fixes when the persistent queue will stop being only an experimental feature.
